rasa/core/processor.py

```diff
@@ -636,12 +634,24 @@ async def _cancel_reminders(
                         scheduler.remove_job(scheduled_job.id)
 
     async def _run_action(
-        self, action, tracker, output_channel, nlg, policy=None, confidence=None
+        self,
+        action,
+        tracker,
+        output_channel,
+        nlg,
+        policy=None,
+        confidence=None,
+        metadata: Optional[Dict] = None,
     ) -> bool:
         # events and return values are used to update
         # the tracker state after an action has been taken
         try:
-            events = await action.run(output_channel, nlg, tracker, self.domain)
+            if action.name() == ACTION_SESSION_START_NAME:
```

Comment:
please add a comment explaining what is going on here (i.e. why this distinction)